src/coreComponents/physicsSolvers/fluidFlow/wells/CompositionalMultiphaseWell.cpp

```diff
@@ -878,14 +935,47 @@ void CompositionalMultiphaseWell::updateTotalMassDensity( WellElementSubRegion &
 
   string const & fluidName = subRegion.getReference< string >( viewKeyStruct::fluidNamesString() );
   MultiFluidBase & fluid = subRegion.getConstitutiveModel< MultiFluidBase >( fluidName );
-
+  fluid.isThermal() ?
+  thermalCompositionalMultiphaseWellKernels::
+    TotalMassDensityKernelFactory::
+    createAndLaunch< parallelDevicePolicy<> >( m_numComponents,
+                                               m_numPhases,
+                                               subRegion,
+                                               fluid )
+  :
   TotalMassDensityKernelFactory::
     createAndLaunch< parallelDevicePolicy<> >( m_numComponents,
                                                m_numPhases,
                                                subRegion,
                                                fluid );
+
 }
+void CompositionalMultiphaseWell::assembleSystem( real64 const time,
```

Comment:
why this becomes different with WellSolverBase::assembleSystem ?

Reply:
Previously well solver base assemble system cold drive both single phase and compositional runs (same calling sequence  assembleAccum, assemblevolbal ....).  The composiitonal Jac assemble has been changed... looks a bit more like the reservoir assemlbeSystem , supports thermal, and the accumulation and vol bal kernels have been combined.. Thermal hasn't been added to single phase yet  so single phase and compositional need there own assembleSystem..

Reply:
i had a second look and the two `assembleSystem` versions for single phase and compositional look the same after your changes
could you please double check that everything is correct?

Reply:
you are correct.  in latest push they are moved from specializations and wellsolverbase has assemble system